jwst/resample/resample_utils.py

```diff
@@ -185,20 +186,83 @@ def build_driz_weight(model, weight_type=None, good_bits=None):
     """
     dqmask = build_mask(model.dq, good_bits)
 
-    if weight_type == 'ivm':
+    if weight_type and weight_type.startswith('ivm'):
+        weight_type = weight_type.strip()
+        selective_median = weight_type.startswith('ivm-smed')
+
+        bitvalue = interpret_bit_flags(good_bits, mnemonic_map=pixel)
+        if bitvalue is None:
+            bitvalue = 0
+        saturation = pixel['SATURATED']
+
+        if selective_median and not (bitvalue & saturation):
+            selective_median = False
+            weight_type = 'ivm'
+
         if (model.hasattr("var_rnoise") and model.var_rnoise is not None and
                 model.var_rnoise.shape == model.data.shape):
             with np.errstate(divide="ignore", invalid="ignore"):
                 inv_variance = model.var_rnoise**-1
+
             inv_variance[~np.isfinite(inv_variance)] = 1
+
+            if weight_type != 'ivm':
+                ny, nx = inv_variance.shape
+
+                # apply a median filter to smooth the weight at saturated
+                # (or high read-out noise) single pixels. keep kernel size
+                # small to still give lower weight to extended CRs, etc.
+                ksz = weight_type[8 if selective_median else 7 :]
+                if ksz:
+                    kernel_size = int(ksz)
+                    if not (kernel_size % 2):
+                        raise ValueError(
+                            'Kernel size of the median filter in IVM weighting'
+                            ' must be an odd integer.'
+                        )
+                else:
+                    kernel_size = 3
+
+                ivm_copy = inv_variance.copy()
+
+                if selective_median:
+                    # apply median filter selectively only at
+                    # points of partially saturated sources:
+                    jumps = np.where(model.dq & saturation)
```

Comment:
I agree. That would be an oversight on my part. Thanks for catching this. Let me check

Reply:
Everything is fine. On line 238 below, this is handled via `dqmask` itself computed on line 187:
```python
data = ivm_copy[y1:y2, x1:x2][dqmask[y1:y2, x1:x2]]
```

Reply:
However, if user says that fully saturated pixels are "good" (i.e., should not be ignored), then it will be exactly the situation that you describe but that's what the user wants...

Reply:
Ah, I see what you mean (initially I thought you were referring to using DO_NOT_USE pixels in the computation of the median). I still do not think this is an issue because as long as `dqmask` is 0, the associated weight for that pixel will be set to 0 on line 260 (`result = inv_variance * dqmask`). However, you are right and thanks for noticing this: it doesn't make sense to compute median if it is going to be ignored.

Reply:
Ah, right, I missed that.  Thumbs up!

Reply:
Latest commit should fix this. Please review